dynamiq/nodes/agents/react.py

```diff
@@ -587,24 +595,33 @@ def generate_structured_output_schemas(self):
         self.format_schema = schema
 
     @staticmethod
-    def filter_format_type(param_type: str) -> str:
-        "Filters proper type for an function calling schema"
-        match param_type:
-            case "bool":
-                return "boolean"
-            case "int":
-                return "integer"
-            case "float":
-                return "float"
-        return "string"
+    def filter_format_type(param_type: str | type) -> str:
+        """Filters proper type for a function calling schema."""
+        if isinstance(param_type, str):
```

Comment:
Also it seems it will never get here, as you are passing class type not str.

Reply:
Maybe just change conditions in order.

Reply:
Should be updated

Reply:
It seems that it still have the same behaviour. Or I miss something.

Reply:
Here are tests results:
```
Testing with string inputs:
Input: 'bool', Output: boolean, Expected: 'boolean'
Input: 'int', Output: integer, Expected: 'integer'
Input: 'float', Output: float, Expected: 'float'
Input: 'some_random_string', Output: string, Expected: 'string'

Testing with Python types:
Input: int, Output: int, Expected: 'int'
Input: str, Output: str, Expected: 'str'
Input: float, Output: float, Expected: 'float'
Input: bool, Output: bool, Expected: 'bool'
Input: list, Output: list, Expected: 'list'
Input: dict, Output: dict, Expected: 'dict'

Testing with Union inputs:
Input: Union[int, str], Output: int, Expected: 'int'
Input: Union[None, list], Output: list, Expected: 'list'
Input: Union[float, None], Output: float, Expected: 'float'

Testing with complex Union inputs:
Input: Union[None, Union[int, str]], Output: int, Expected: 'int'
Input: Union[Union[str, int], list], Output: str, Expected: 'str'

Testing with fallback cases:
Input: CustomType, Output: CustomType, Expected: 'CustomType'
Input: object, Output: object, Expected: 'object'

Testing with edge cases:
Input: None, Output: string, Expected: 'string'
Input: 123, Output: string, Expected: 'string'
```

Reply:
Yes, but here you pass self.filter_format_type(field.annotation) not string.

Reply:
Yes but expected for (Input: int, Output: int, Expected: 'int') Expected should be 'integer' not int.

Reply:
Also it seems we never pass string to a function  (you pass filter_format_type(field.annotation))

Reply:
```
type_mapping = {
    int: "integer",
    float: "float",
    bool: "boolean",
    str: "string",
    list: "list",
    dict: "dict",
}
```
This mapping can help address naming issues, so please double-check it. Regarding strings input, it would be good to keep this behavior.

Reply:
`type_mapping = {
    int: "integer",
    float: "number",
    bool: "boolean",
    str: "string",
}
`
Regarding list and dict, it has to map to `array` and `object`, but we also need to define special structured for them. So their behaviour would be different from first 4. I guess it is fine now to have only first 4 for now.